src/ripple/protocol/impl/tokens.cpp

```diff
@@ -241,5 +261,358 @@ decodeBase58Token(std::string const& s, TokenType type)
     // Skip the leading type byte and the trailing checksum.
     return ret.substr(1, ret.size() - 1 - guard.size());
 }
+}  // namespace b58_ref
+
+#ifndef _MSC_VER
+namespace b58_fast {
+namespace detail {
+B58Result<std::span<std::uint8_t>>
+b256_to_b58(std::span<std::uint8_t const> input, std::span<std::uint8_t> out)
+{
+    // Allocate enough base 58^10 coeff for encoding 38 bytes
+    // (33 bytes for nodepublic + 1 byte token + 4 bytes checksum)
+    // log(2^(38*8),58^10)) ~= 5.18. So 6 coeff are enough
+    if (input.size() > 38)
+    {
+        return Unexpected(TokenCodecErrc::InputTooLarge);
+    };
+
+    auto count_leading_zeros = [&](auto const& col) -> std::size_t {
+        std::size_t count = 0;
+        for (auto const& c : col)
+        {
+            if (c != 0)
+            {
+                return count;
+            }
+            count += 1;
+        }
+        return count;
+    };
+
+    auto const input_zeros = count_leading_zeros(input);
+    input = input.subspan(input_zeros);
+
+    std::array<std::uint64_t, 6> base_58_10_coeff{};
+    std::array<std::uint64_t, 5> base_2_64_coeff_buf{};
+    std::span<std::uint64_t> const base_2_64_coeff =
+        [&]() -> std::span<std::uint64_t> {
+        // convert from input from big endian to native u64, lowest coeff first
+        std::size_t num_coeff = 0;
+        for (int i = 0; i < 5; ++i)
+        {
+            if (i * 8 > input.size())
```

Comment:
This can be `>=` right? Right now, `i * 8` can equal `input.size()`, which means the loop on lines 319-323 may be a no-op. Changing the condition means the loop can assume the range `[0, src_i_end)` is non-empty, and that the assignment on line 324 will not execute when `i * 8 == input.size()`. I think that change in behavior is ok.

Reply:
Yes, agreed that `>=` works and is better. Fixed in 3c80d5304e [fold] Address misc PR issues